src/js/controller/captions.js

```diff
@@ -152,8 +154,8 @@ const Captions = function(_model) {
     function _setCaptionsList() {
         const captionsList = _captionsMenu();
         if (listIdentity(captionsList) !== listIdentity(_model.get('captionsList'))) {
-            _selectDefaultIndex(_defaultIndex);
             _model.set('captionsList', captionsList);
+            _selectDefaultIndex(_defaultIndex);
```

Comment:
This change will change event order which we should avoid.

 `_selectDefaultIndex ` calls `_setCurrentIndex` at the end. Can we split these apart so that we maintain current event order?

It would go something like:
```js
            const captionsMenuIndex = _selectDefaultIndex(_defaultIndex); // does not call _setCurrentIndex
            _model.set('captionsList', captionsList);
            _setCurrentIndex(captionsMenuIndex);
```
And below:
```js
this.selectDefaultIndex = (defaultIndex) =>
            const captionsMenuIndex = _selectDefaultIndex(defaultIndex);
            _setCurrentIndex(captionsMenuIndex);
};

Reply:
hmmmm... that would still change event order. How does the order prevent us from sorting side-loaded last? I'm a little unclear on this.

Reply:
This is fixing another issue, not the side-loaded captions - this line is causing js exception when we have sideloaded captions. Here is the setup that breaks on last successful [player-develop](http://player-develop-test-jenkins.longtailvideo.com/builds/lastSuccessfulBuild/archive/test/harness.html?edition=ads):
```
{
  "file": "http://wowzaec2demo.streamlock.net/vod-multitrack/_definst_/smil:ElephantsDream/elephantsdream2.smil/playlist.m3u8",
  "tracks": [
      {
        "file": "//playertest.longtailvideo.com/assets/os/captions/bunny-en.vtt",
        "label": "Side-loaded",
        "kind": "captions"
      },
      {
            "file": "//playertest.longtailvideo.com/assets/os/captions/bunny-en.srt",
            "kind": "captions",
            "label": "English"
        }
 
  ],
  "analytics": {
    "client": "../bin-debug/jwpsrv.js",
    "debug": true
  }
}
```

Reply:
The problem here is that when the side-loaded captions are first loaded (without the playlistItem captions), the `defaultIndex` is greater than the size of the `captionsList`